File: typescript/base-hmi-types/src/screens/controls/HmiRadarChartControl.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiProperty } from "../base/HmiProperty.js";
import { HmiColor } from "../base/HmiColor.js";
import { HmiFont } from "../base/HmiFont.js";
import { HmiLineStyle } from "../base/HmiLineStyle.js";
import { HmiControlWindowBase } from "../base/HmiControlWindowBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiRadarLegendPosition } from "./HmiRadarLegendPosition.js";
import { HmiRadarShape } from "./HmiRadarShape.js";
import { HmiRadarChartCategory } from "./HmiRadarChartCategory.js";
import { HmiRadarChartSeries } from "./HmiRadarChartSeries.js";

/**
 * Represents a radar/spider chart control. Product-specific chart payloads can
 * remain attached through the inherited source-data fields until decoded.
 */
export class HmiRadarChartControl extends HmiControlWindowBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiRadarChartControl;
  }

  title?: HmiMultilingualText;
  seriesCount?: HmiProperty<number>;
  categoryCount?: HmiProperty<number>;
  radarShape?: HmiProperty<HmiRadarShape>;
  sourceRadarShape?: string;
  chartBackgroundColor?: HmiProperty<HmiColor>;
  gridLineStyle?: HmiProperty<HmiLineStyle>;
  sourceGridLineStyle?: string;
  gridLineColor?: HmiProperty<HmiColor>;
  bandedColor?: HmiProperty<HmiColor>;
  showLegend?: HmiProperty<boolean>;
  legendPosition?: HmiProperty<HmiRadarLegendPosition>;
  sourceLegendPosition?: string;
  decimalPlaces?: HmiProperty<number>;
  refreshRateSeconds?: HmiProperty<number>;
  titleFont?: HmiFont;
  categoryFont?: HmiFont;
  legendFont?: HmiFont;
  dataLabelFont?: HmiFont;
  readonly categories: HmiRadarChartCategory[] = [];
  readonly series: HmiRadarChartSeries[] = [];
}

File: src/BaseHmiTypes/Screens/Controls/HmiRadarChartControl.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

/// <summary>
/// Represents a radar/spider chart control. Product-specific chart payloads can
/// remain attached through the inherited source-data fields until decoded.
/// </summary>
public class HmiRadarChartControl : HmiControlWindowBase
{
    public HmiRadarChartControl()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiRadarChartControl;
    }

    public HmiMultilingualText? Title { get; set; }

    public HmiProperty<int>? SeriesCount { get; set; }

    public HmiProperty<int>? CategoryCount { get; set; }

    public HmiProperty<HmiRadarShape>? RadarShape { get; set; }

    public string? SourceRadarShape { get; set; }

    public HmiProperty<HmiColor>? ChartBackgroundColor { get; set; }

    public HmiProperty<HmiLineStyle>? GridLineStyle { get; set; }

    public string? SourceGridLineStyle { get; set; }

    public HmiProperty<HmiColor>? GridLineColor { get; set; }

    public HmiProperty<HmiColor>? BandedColor { get; set; }

    public HmiProperty<bool>? ShowLegend { get; set; }

    public HmiProperty<HmiRadarLegendPosition>? LegendPosition { get; set; }

    public string? SourceLegendPosition { get; set; }

    public HmiProperty<int>? DecimalPlaces { get; set; }

    public HmiProperty<double>? RefreshRateSeconds { get; set; }

    public HmiFont? TitleFont { get; set; }

    public HmiFont? CategoryFont { get; set; }

    public HmiFont? LegendFont { get; set; }

    public HmiFont? DataLabelFont { get; set; }
}
